extensions/Lily/MoreEvents.js

```diff
@@ -0,0 +1,424 @@
+    { text: 'left arrow', value: 'left arrow' },
+    { text: 'enter', value: 'enter' },
+    // TW: Extra Special Inputs
+    { text: 'backspace', value: 'backspace' },
+    { text: 'delete', value: 'delete' },
+    { text: 'shift', value: 'shift' },
+    { text: 'caps lock', value: 'caps lock' },
+    { text: 'scroll lock', value: 'scroll lock' },
+    { text: 'control', value: 'control' },
+    { text: 'escape', value: 'escape' },
+    { text: 'insert', value: 'insert' },
+    { text: 'home', value: 'home' },
+    { text: 'end', value: 'end' },
+    { text: 'page up', value: 'page up' },
+    { text: 'page down', value: 'page down' },
+    // Letter Keyboard Inputs
+    { text: 'a', value: 'a' },
+    { text: 'b', value: 'b' },
+    { text: 'c', value: 'c' },
+    { text: 'd', value: 'd' },
+    { text: 'e', value: 'e' },
+    { text: 'f', value: 'f' },
+    { text: 'g', value: 'g' },
+    { text: 'h', value: 'h' },
+    { text: 'i', value: 'i' },
+    { text: 'j', value: 'j' },
+    { text: 'k', value: 'k' },
+    { text: 'l', value: 'l' },
+    { text: 'm', value: 'm' },
+    { text: 'n', value: 'n' },
+    { text: 'o', value: 'o' },
+    { text: 'p', value: 'p' },
+    { text: 'q', value: 'q' },
+    { text: 'r', value: 'r' },
+    { text: 's', value: 's' },
+    { text: 't', value: 't' },
+    { text: 'u', value: 'u' },
+    { text: 'v', value: 'v' },
+    { text: 'w', value: 'w' },
+    { text: 'x', value: 'x' },
+    { text: 'y', value: 'y' },
+    { text: 'z', value: 'z' },
+    // Number Keyboard Inputs
+    { text: '0', value: '0' },
+    { text: '1', value: '1' },
+    { text: '2', value: '2' },
+    { text: '3', value: '3' },
+    { text: '4', value: '4' },
+    { text: '5', value: '5' },
+    { text: '6', value: '6' },
+    { text: '7', value: '7' },
+    { text: '8', value: '8' },
+    { text: '9', value: '9' },
+  ];
+
+  setInterval(() => {
+    runtime.startHats('lmsMoreEvents_always', {
+      CONDITION: 'true'
+    });
+
+    runtime.startHats('lmsMoreEvents_whileTurboMode', {
+      STATE: (runtime.turboMode) ? 'enabled' : 'disabled'
+    });
+  });
+
+  var lastValues = {};
+
+  class MoreEvents {
+    constructor() {
+      // Stop Sign Clicked contributed by @CST1229
+      runtime.shouldExecuteStopClicked = true;
+      runtime.on('BEFORE_EXECUTE', () => {
+        runtime.shouldExecuteStopClicked = false;
+      });
+      runtime.on('PROJECT_STOP_ALL', () => {
+        if (runtime.shouldExecuteStopClicked)
+          queueMicrotask(() => runtime.startHats('lmsMoreEvents_whenStopClicked'));
+      });
+      runtime.on('AFTER_EXECUTE', () => {
+        runtime.shouldExecuteStopClicked = true;
+      });
+      const originalGreenFlag = vm.greenFlag;
+      vm.greenFlag = function() {
+        runtime.shouldExecuteStopClicked = false;
+        originalGreenFlag.call(this);
+      };
+
+      runtime.on('STAGE_SIZE_CHANGED', () => {
+        runtime.startHats('lmsMoreEvents_whenRuntimeOptionChanged', {

```

Comment:
should these not be in runtime options

Reply:
I can remove them and put them into a separate PR later, if that's preferred. I'm inclined to agree